exporter/opentelemetry-exporter-zipkin/test-requirements.txt

```diff
@@ -0,0 +1,22 @@
+asgiref==3.7.2
+attrs==23.2.0
+Deprecated==1.2.14
+flaky==3.7.0
+importlib-metadata==6.11.0
+iniconfig==2.0.0
+packaging==23.2
+pluggy==1.4.0
+py==1.11.0
+py-cpuinfo==9.0.0
+pytest==7.1.3
+pytest-benchmark==4.0.0
+tomli==2.0.1
+typing_extensions==4.10.0
+wrapt==1.16.0
+zipp==3.17.0
+-e opentelemetry-api
+-e exporter/opentelemetry-exporter-zipkin-json
```

Comment:
I think you need the `pip freeze` from each of the editable installs as well. Dependencies like [requests and protobuf](https://github.com/open-telemetry/opentelemetry-python/blob/main/exporter/opentelemetry-exporter-zipkin-proto-http/pyproject.toml#L31) might need to be added.

Reply:
Not necessarily.

Actually, `opentelemetry-exporter-zipkin-proto-http` should not have `protobuf` as one of its dependencies:

```
tigre@hilleman:~/github/ocelotl/opentelemetry-python/exporter/opentelemetry-exporter-zipkin-proto-http$ git lg | head -n1
* 190f3cc08 -  (HEAD -> main, upstream/main, origin/main, origin/HEAD) Update pyproject.toml (#3576) (70 minutes ago) <Fritz Stapfer Paz>
tigre@hilleman:~/github/ocelotl/opentelemetry-python/exporter/opentelemetry-exporter-zipkin-proto-http$ ack protobuf
pyproject.toml
31:  "protobuf ~= 3.12",

src/opentelemetry/exporter/zipkin/proto/http/__init__.py
26:collector endpoint using HTTP and supports v2 protobuf.

src/opentelemetry/exporter/zipkin/proto/http/v2/__init__.py
44:        return "application/x-protobuf"

src/opentelemetry/exporter/zipkin/proto/http/v2/gen/zipkin_pb2.py
5:from google.protobuf import descriptor as _descriptor
6:from google.protobuf import message as _message
7:from google.protobuf import reflection as _reflection
8:from google.protobuf import symbol_database as _symbol_database

src/opentelemetry/exporter/zipkin/proto/http/v2/gen/zipkin_pb2.pyi
3:from google.protobuf.descriptor import (
4:    Descriptor as google___protobuf___descriptor___Descriptor,
5:    EnumDescriptor as google___protobuf___descriptor___EnumDescriptor,
6:    FileDescriptor as google___protobuf___descriptor___FileDescriptor,
9:from google.protobuf.internal.containers import (
10:    RepeatedCompositeFieldContainer as google___protobuf___internal___containers___RepeatedCompositeFieldContainer,
13:from google.protobuf.message import (
14:    Message as google___protobuf___message___Message,
45:DESCRIPTOR: google___protobuf___descriptor___FileDescriptor = ...
47:class Span(google___protobuf___message___Message):
48:    DESCRIPTOR: google___protobuf___descriptor___Descriptor = ...
52:        DESCRIPTOR: google___protobuf___descriptor___EnumDescriptor = ...
75:    class TagsEntry(google___protobuf___message___Message):
76:        DESCRIPTOR: google___protobuf___descriptor___Descriptor = ...
111:    def annotations(self) -> google___protobuf___internal___containers___RepeatedCompositeFieldContainer[type___Annotation]: ...
142:class Endpoint(google___protobuf___message___Message):
143:    DESCRIPTOR: google___protobuf___descriptor___Descriptor = ...
165:class Annotation(google___protobuf___message___Message):
166:    DESCRIPTOR: google___protobuf___descriptor___Descriptor = ...
184:class ListOfSpans(google___protobuf___message___Message):
185:    DESCRIPTOR: google___protobuf___descriptor___Descriptor = ...
188:    def spans(self) -> google___protobuf___internal___containers___RepeatedCompositeFieldContainer[type___Span]: ...
203:class ReportResponse(google___protobuf___message___Message):
204:    DESCRIPTOR: google___protobuf___descriptor___Descriptor = ...
```

That's not the same with `requests`:

```
tigre@hilleman:~/github/ocelotl/opentelemetry-python/exporter/opentelemetry-exporter-zipkin-proto-http$ git lg | head -n1
* 190f3cc08 -  (HEAD -> main, upstream/main, origin/main, origin/HEAD) Update pyproject.toml (#3576) (71 minutes ago) <Fritz Stapfer Paz>
tigre@hilleman:~/github/ocelotl/opentelemetry-python/exporter/opentelemetry-exporter-zipkin-proto-http$ ack requests
tests/test_zipkin_exporter.py
20:import requests
62:        self.assertIsInstance(exporter.session, requests.Session)
91:        self.assertIsInstance(exporter.session, requests.Session)
112:        session_param = requests.Session()
125:        self.assertIsInstance(exporter.session, requests.Session)
139:    @patch("requests.Session.post")
147:    @patch("requests.Session.post")
155:    @patch("requests.Session.post")
171:    @patch("requests.Session.post")
184:    @patch("requests.Session.post")

pyproject.toml
32:  "requests ~= 2.7",

src/opentelemetry/exporter/zipkin/proto/http/__init__.py
34:    import requests
53:        # session=requests.Session()
78:import requests
105:        session: Optional[requests.Session] = None,
136:        self.session = session or requests.Session()
```

Nevertheless, we don't need `requests` (or `protobuf`) as a dependency in `exporter/opentelemetry-exporter-zipkin/test-requirements.txt` because the tests for `opentelemetry-exporter-zipkin` don't require it (since they all pass without `requests` or `protobuf` being installed).